src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FFIDMetadataFields.scala

```diff
@@ -0,0 +1,45 @@
+package uk.gov.nationalarchives.tdr.api.graphql.fields
+
+import java.util.UUID
+
+import sangria.schema.{Argument, Field, InputObjectType, ObjectType, fields}
+import sangria.macros.derive._
+import FieldTypes._
+import sangria.marshalling.circe._
+import io.circe.generic.auto._
+import uk.gov.nationalarchives.tdr.api.auth.ValidateHasFFIDMetadataAccess
+import uk.gov.nationalarchives.tdr.api.graphql.ConsignmentApiContext
+
+object FFIDMetadataFields {
+  case class FFIDMetadata(fileId: UUID,
+                          software: String,
+                          softwareVersion: String,
+                          binarySignatureFileVersion: String,
+                          containerSignatureFileVersion: String,
+                          method: String,
+                          extension: Option[String] = None,
+                          identificationBasis: String,
+                          puid: Option[String],
+                          datetime: Long)
```

Comment:
Is this the timestamp that we think is for auditing? If so, should it be generated by the API so that the API is responsible for all the integrity data, rather than trusting the client application?

Reply:
Normally yes but the reason I did it like this is that @Dclipsham said that we would need the scan date from the siegfried result so that's why I've passed it in.

Reply:
I've changed it now so the API generates the datetime.

Reply:
Yeah, I'd have thought the date we're interested in is when the data is saved, since that's when it becomes available to other parts of the system. Do you know if that's good enough for David's purposes, or is there some other reason to save the scan date?

Just to check - will the Siegfried version information that we're saving tell us what version of the file format signatures is being used? Assuming that's the important thing if we need to debug any unexpected results?

Reply:
Yep, that's what the binarySignatureFileVersion and containerSignatureFileVersion do. There's a conversation on slack recently where I've asked what he needs. And for the scan date, I can't see any reason why the few minutes it takes to process through the system is going to make much difference unless people are uploading at 2 minutes to midnight which seems unlikely.